File: xml/java/update/xml_update.java
// --------------------------------------------------------------
/*
	xml_file/java/update/xml_update.java

				May/26/2011

*/
// --------------------------------------------------------------
import	java.util.HashMap;

// --------------------------------------------------------------
public class xml_update {
// --------------------------------------------------------------
public static void main(String[] args) throws Exception
{
	System.out.println("*** 開始 *** xml_update ***");
	String xml_file = args[0];

	String	id = args[1];
	int	population = Integer.parseInt (args[2]);
	System.out.print ("\tid = " + id);
	System.out.println ("\tpopulation = " + population);

	String str_xml = "";
	try
	{
	str_xml = file_io.file_to_str_proc (xml_file);
	}
	catch (Exception ee)
	{
	ee.printStackTrace ();
	}

	HashMap <String, HashMap<String,String>> dict_aa
		= xml_manipulate.xml_to_dict_proc (str_xml);

	text_manipulate.dict_update_proc (dict_aa,id,population);

	String xml_str_new = xml_manipulate.dict_to_xml_proc (dict_aa);

	file_io.file_write_proc (xml_file,xml_str_new);


	System.out.println("***** 終了 *****");
}
 
// --------------------------------------------------------------
}
// --------------------------------------------------------------

File: xml/python/update/xml_update.py
#! /usr/bin/python3
# -*- coding: utf-8 -*-
#
#	python/update/xml_update.py
#
#						Jul/25/2014
#
# --------------------------------------------------------------------
import	xmltodict
import	dicttoxml
import	sys
import	string
sys.path.append ('/var/www/data_base/common/python_common')
#
from file_io import file_to_str_proc
from file_io import file_write_proc
from text_manipulate import dict_update_proc
#
# --------------------------------------------------------------------
sys.stderr.write("*** 開始 ***\n")
#
xml_file = sys.argv[1]
key_in = sys.argv[2]
population_in = int (sys.argv[3])
print ("%s\t%d" % (key_in, population_in))
#
xml_str=file_to_str_proc (xml_file)
dict_aa = xmltodict.parse(xml_str)['root']
#
dict_update_proc (dict_aa,key_in,population_in)
#
byte_str = dicttoxml.dicttoxml(dict_aa,attr_type=False)
out_str = byte_str.decode('utf8')
file_write_proc (xml_file,out_str)
#
sys.stderr.write("*** 終了 ***\n")
# --------------------------------------------------------------------
